app/model/sdl/VehicleInfoModel.js

```diff
@@ -403,27 +403,27 @@ SDL.SDLVehicleInfoModel = Em.Object.create(
     CalculateSubscriptionsResultCode: function(subscriptions) {
       var statistic = {
         success_count: 0,
-        total_count: 0
+        total_count: 0,
+        ignore: 0
       };
 
-      var code = SDL.SDLModel.data.resultCode.SUCCESS;
       for (var key in subscriptions) {
         statistic.total_count++;
         if (subscriptions[key].resultCode == 'SUCCESS') {
           statistic.success_count++;
         }
-        else if (subscriptions[key].resultCode == 'VEHICLE_DATA_NOT_AVAILABLE') {
-          code = SDL.SDLModel.data.resultCode.DATA_NOT_AVAILABLE;
-        }
-        else if (code == SDL.SDLModel.data.resultCode.SUCCESS && 
-            (subscriptions[key].resultCode == 'DATA_ALREADY_SUBSCRIBED' || 
-             subscriptions[key].resultCode == 'DATA_NOT_SUBSCRIBED')) {
-          code = SDL.SDLModel.data.resultCode.IGNORED;
+        else if (subscriptions[key].resultCode == 'DATA_ALREADY_SUBSCRIBED' ||
+             subscriptions[key].resultCode == 'DATA_NOT_SUBSCRIBED') {
+          statistic.ignore++;
         }
       }
 
-      if (statistic.success_count > 0) {
-        return SDL.SDLModel.data.resultCode.WARNINGS;
+      var code = SDL.SDLModel.data.resultCode.WARNINGS;
+      if (statistic.total_count == 0 || statistic.ignore == statistic.total_count) {
+        code = SDL.SDLModel.data.resultCode.IGNORED;
+      }
+      else if (statistic.total_count == statistic.success_count) {
+        code = SDL.SDLModel.data.resultCode.SUCCESS;
```

Comment:
@mkorniichuk as this is a result code, would be simpler to do:
```suggestion
    if (statistic.total_count == 0 || statistic.ignore == statistic.total_count) {
       return SDL.SDLModel.data.resultCode.IGNORED;
    }
    
    if (statistic.total_count == statistic.success_count) {
      return SDL.SDLModel.data.resultCode.SUCCESS;
    }
    
    return SDL.SDLModel.data.resultCode.WARNINGS;    
```

Reply:
@AKalinich-Luxoft good point. Fixed in 061f5c11beaf6667108df20e261da1394b28a6b0